src/test/java/core/basesyntax/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,77 @@
+package core.basesyntax;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.User;
+import core.basesyntax.service.RegistrationService;
+import core.basesyntax.service.RegistrationServiceImpl;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+public class RegistrationServiceImplTest {
+    private static RegistrationService registrationService;
+    private static User user;
+
+    @BeforeAll
+    public static void setUp() {
+        registrationService = new RegistrationServiceImpl();
+    }
+
+    @BeforeEach
+    public void userInit() {
+        user = new User(12121212L, "testUser", "123user", 20);
+    }
+
+    @Test
+    public void register_validUser_Ok() {
+        User actual = registrationService.register(user);
+        assertEquals(user, actual);
+    }
+
+    @Test
+    public void register_nullUser_notOk() {
+        assertThrows(RuntimeException.class, () -> registrationService.register(null));
+    }
+
+    @Test
+    public void register_loginExists_notOk() {
+        User user2 = new User(12958212L, "testUser", "123user123", 47);
+        registrationService.register(user);
+        assertThrows(RuntimeException.class, () -> registrationService.register(user2));
+    }
+
+    @Test
+    public void register_nullLogin_notOk() {
+        user.setLogin(null);
+        assertThrows(RuntimeException.class, () -> registrationService.register(user));
+    }
+
+    @Test
+    public void register_invalidPasswordLength_notOk() {
+        user.setPassword("1234");
+        assertThrows(RuntimeException.class, () -> registrationService.register(user));
+    }
+
+    @Test
+    public void register_nullPassword_notOk() {
+        user.setPassword(null);
+        assertThrows(RuntimeException.class, () -> registrationService.register(user));
+    }
+
+    @Test
+    public void register_invalidAge_notOk() {
+        user.setAge(15);
+        assertThrows(RuntimeException.class, () -> registrationService.register(user));
+        user.setAge(null);
+        assertThrows(RuntimeException.class, () -> registrationService.register(user));
+    }
+
```

Comment:
add tests that will check valid password, valid age and null age

Reply:
but all valid data is checking in register_validUser_Ok()

Reply:
Sorry, my mistake